processor/src/main/java/org/mapstruct/ap/MappingProcessor.java

```diff
@@ -163,6 +164,32 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
             Set<TypeElement> mappers = getMappers( annotations, roundEnvironment );
             processMapperElements( mappers, roundContext );
         }
+        else if ( !deferredMappers.isEmpty() ) {
+            // If the processing is over and there are deferred mappers it means something wrong occurred and
+            // MapStruct didn't generate implementations for those
+            for ( DeferredMapper deferredMapper : deferredMappers ) {
+
+                TypeElement deferredMapperElement = deferredMapper.deferredMapperElement;
+                Element erroneousElement = deferredMapper.erroneousElement;
+                String erroneousElementName;
+
+                if ( erroneousElement instanceof QualifiedNameable ) {
+                    erroneousElementName = ( (QualifiedNameable) erroneousElement ).getQualifiedName().toString();
+                }
+                else {
+                    erroneousElementName = erroneousElement.getSimpleName().toString();
+                }
+
+                processingEnv.getMessager()
```

Comment:
is there any way to figure out what annotatiotion processors are in the compilation cycle? If so, it we should list them. If not: it seems that the error message needs to point more clearly to the underlying cause in a more generic way. something along the lines what you have now, but a hint that another ap (not generating classes) might be the problem.

Reply:
I don't think that we get can the current annotation processors. But if you have some better idea for the message I would gladly hear it. 

I can add the hint that maybe some another annotation processor didn't create the class and that you could enable verbose mode for mapstruct so you can see what is happening.

Reply:
I changed the text a bit let me know what you think